packages/@aws-cdk/aws-eks/lib/manifest-resource/index.py

```diff
@@ -0,0 +1,124 @@
+import subprocess
+import os
+import json
+import logging
+import yaml
+import boto3
+from uuid import uuid4
+from botocore.vendored import requests
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+# these are coming from the kubectl layer
+os.environ['PATH'] = '/opt/kubectl:/opt/awscli:' + os.environ['PATH']
+
+outdir = os.environ.get('TEST_OUTDIR', '/tmp')
+kubeconfig = os.path.join(outdir, 'kubeconfig')
+
+CFN_SUCCESS = "SUCCESS"
+CFN_FAILED = "FAILED"
+
+def handler(event, context):
+
+    def cfn_error(message=None):
+        logger.error("| cfn_error: %s" % message)
+        cfn_send(event, context, CFN_FAILED, reason=message)
+
+    try:
+        logger.info(json.dumps(event))
+
+        request_type = event['RequestType']
+        props = event['ResourceProperties']
+        old_props = event.get('OldResourceProperties', {})
+        physical_id = event.get('PhysicalResourceId', None)
+        manifest_text = props['Manifest']
+
+        cluster_name = os.environ.get('CLUSTER_NAME', None)
+        if cluster_name is None:
+            cfn_error("CLUSTER_NAME is missing in environment")
+            return
+
+        # "log in" to the cluster
+        subprocess.check_call([ 'aws', 'eks', 'update-kubeconfig',
+            '--name', cluster_name,
+            '--kubeconfig', kubeconfig
+        ])
+
+        # convert manifest to yaml and write to file
+        manifest_yaml = yaml.dump_all(json.loads(manifest_text))
+        logger.info(manifest_yaml)
+        manifest_file = os.path.join(outdir, 'manifest.yaml')
+        with open(manifest_file, "w") as f:
+            f.write(manifest_yaml)
+        logger.info("manifest written to: %s" % manifest_file)
+
+        if request_type == 'Create' or request_type == 'Update':
+            kubectl('apply', manifest_file)
+        elif request_type == "Delete":
+            # TODO: never fail
+            try:
+                kubectl('delete', manifest_file)
+            except Exception as e:
+                logger.info("delete error: %s" % e)
+
+        # if we are creating a new resource, allocate a physical id for it
+        # otherwise, we expect physical id to be relayed by cloudformation
+        if request_type == 'Create':
+            physical_id = "%s/%s" % (cluster_name, str(uuid4()))
```

Comment:
Kubernetes is going to create a unique ID for this as well. Is there any value add in making them match? I think immediately after apply you can look this up.

Reply:
After looking at the tests, are we submitting more than resource at a time? If so then obviously my physical ID comment is not possible.

Reply:
Yes, this allows submitting a manifest with multiple resources. Seems like it's a common pattern in k8s, but I am not 100% sure it's needed. What do you think? How do you look up the ID? isn't that the `metadata.name` of the resource?

Reply:
```$> kubectl get svc -o yaml
apiVersion: v1
items:
- apiVersion: v1
  kind: Service
  metadata:
    creationTimestamp: "2019-07-29T05:40:03Z"
    labels:
      component: apiserver
      provider: kubernetes
    name: kubernetes
    namespace: default
    resourceVersion: "6"
    selfLink: /api/v1/namespaces/default/services/kubernetes
    uid: 5003bd44-b1c3-11e9-b1d6-02978cd9b20e
  spec:
    clusterIP: 172.20.0.1
    ports:
    - name: https
      port: 443
      protocol: TCP
      targetPort: 443
    sessionAffinity: None
    type: ClusterIP
  status:
    loadBalancer: {}
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

It is in the get response above as metadata.uid. After reading comments like this: https://github.com/kubernetes/kubernetes/pull/76870 I think my suggestion is a bad idea. 1) I don't have a clear use cases for making them match other than not generating a unique ID 2) it's unclear to me what happens if they get out of sync. 

I would vote for resolving this convo and implementing this feature when a use case appears. I don't see any reason this isn't a 2-way door. 

Separately we do metadata annotate the CFN stacks, I'm not sure if there is value in that here: [annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/).

Reply:
Ok resolving for now. Thanks for the details